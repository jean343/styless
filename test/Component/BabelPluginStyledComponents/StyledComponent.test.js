import React from 'react';
import renderer from 'react-test-renderer';
import styled, {css} from "styled-components";
import 'jest-styled-components';

test('Basic css prop', () => {
    // https://www.styled-components.com/docs/api#css-prop
    expect(renderer.create(<div
        css={`
            background: papayawhip;
            color: ${"green"};
        `}
    />).toJSON()).toMatchSnapshot();
    expect(renderer.create(<button
        css="padding: 0.5em 1em;"
    />).toJSON()).toMatchSnapshot();

    expect(renderer.create(<div
        css={css`
            background: papayawhip;
            color: ${"green"};
        `}
    />).toJSON()).toMatchSnapshot();
    expect(renderer.create(<button
        css={css`padding: 0.5em 1em;`}
    />).toJSON()).toMatchSnapshot();
});

test('Styless css prop', () => {
    // https://www.styled-components.com/docs/api#css-prop
    expect(renderer.create(<div
        css={`
            background: papayawhip;
            color: ${"green"};
        `}
    />).toJSON()).toMatchSnapshot();
    expect(renderer.create(<button
        css="padding: 0.5em 1em;"
    />).toJSON()).toMatchSnapshot();

    expect(renderer.create(<div
        css={css`
            background: papayawhip;
            color: @color;
        `}
        color="blue"
    />).toJSON()).toMatchSnapshot();
    expect(renderer.create(<button
        css={css`padding: 0.5em 1em;`}
    />).toJSON()).toMatchSnapshot();
});