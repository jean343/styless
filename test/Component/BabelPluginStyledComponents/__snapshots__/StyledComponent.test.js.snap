// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic css prop 1`] = `
.c0 {
  background: papayawhip;
  color: green;
}

<div
  className="c0"
/>
`;

exports[`Basic css prop 2`] = `
.c0 {
  padding: 0.5em 1em;
}

<button
  className="c0"
/>
`;

exports[`Basic css prop 3`] = `
.c0 {
  background: papayawhip;
  color: green;
}

<div
  className="c0"
/>
`;

exports[`Basic css prop 4`] = `
.c0 {
  padding: 0.5em 1em;
}

<button
  className="c0"
/>
`;
